header-footer-grid/Core/Components/PaletteSwitch.php

```diff
@@ -0,0 +1,395 @@
+<?php
+/**
+ * Frontend Palette Switch Component class for Header Footer Grid.
+ *
+ * Name:    Header Footer Grid
+ * Author:  Bogdan Preda <bogdan.preda@themeisle.com>
+ *
+ * @version 1.0.0
+ * @package HFG
+ */
+
+namespace HFG\Core\Components;
+
+use HFG\Core\Settings\Manager as SettingsManager;
+use HFG\Main;
+use Neve\Core\Settings\Config;
+use Neve\Core\Styles\Dynamic_Selector;
+
+/**
+ * Class PaletteSwitch
+ *
+ * @package HFG\Core\Components
+ */
+class PaletteSwitch extends Abstract_Component {
+
+	const COMPONENT_ID      = 'header_palette_switch';
+	const LIGHT_PALETTE_ID  = 'light_palette';
+	const DARK_PALETTE_ID   = 'dark_palette';
+	const TOGGLE_ICON_ID    = 'toggle_icon';
+	const PLACEHOLDER_ID    = 'placeholder';
+	const AUTO_ADJUST       = 'auto_adjust_color';
+	const DESCRIPTION_ID    = 'description_info';
+	const SIZE_ID           = 'icon_size';
+	const DEFAULT_ICON_SIZE = 16;
+
+	/**
+	 * The section icon.
+	 *
+	 * @access protected
+	 * @var string $icon
+	 */
+	protected $icon = 'buddicons-topics';
+	/**
+	 * The component default width.
+	 *
+	 * @since   1.0.0
+	 * @access  protected
+	 * @var int $width
+	 */
+	protected $width = 1;
+	/**
+	 * The component slug.
+	 *
+	 * @since   1.0.0
+	 * @access  protected
+	 * @var string $section
+	 */
+	protected $component_slug = 'hfg-palette-icon';
+
+	/**
+	 * PaletteSwitch constructor
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$this->set_property( 'label', __( 'Palette Switch', 'neve' ) );
+		$this->set_property( 'id', $this->get_class_const( 'COMPONENT_ID' ) );
+		$this->set_property( 'default_selector', '.builder-item--' . $this->get_id() );
+
+		add_filter( 'neve_after_css_root', [ $this, 'toggle_css' ], 10, 1 );
+		add_action( 'wp_enqueue_scripts', [ $this, 'load_scripts' ] );
+	}
+
+	/**
+	 * Load Component Scripts
+	 *
+	 * @return void
+	 */
+	public function load_scripts() {
+		wp_add_inline_script( 'neve-script', $this->toggle_script() );
+	}
+
+	/**
+	 * Get JS contents from file to use as inline script
+	 *
+	 * @return string
+	 */
+	public function toggle_script() {
+		$script = '';
+		global $wp_filesystem;
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		WP_Filesystem();
+		$file            = NEVE_MAIN_DIR . 'header-footer-grid/assets/js/components/palette_switch.js';
```

Comment:
if the file < 1KB minified, we can drop it minified as an inline script, otherwise, we can enqueue it in the footer with wp_script_add_data as async.

Reply:
I've included the minified `js` directly here: https://github.com/Codeinwp/neve/pull/2783/commits/0254586b62881803b499373dea5411a6fbbeb40d#diff-6f8db8b20976151ccb70279d3bf5e266944be7f0f0eb6375d020878468b7c53dR89